File: 1897. Redistribute Characters to Make All Strings Equal/solution.cpp
class Solution {
public:
  bool makeEqual(vector<string>& words) {
    unordered_map<char, int> dictionary;
    for (int index = 0; index < words.size(); index++){
      string word = words[index];
      for (int index2 = 0; index2 < word.size(); index2++){
        char letter = word[index2];
        dictionary[letter]++;
      }
    }
    
    for (auto letter : dictionary){
      if (letter.second % words.size() != 0){
        return false;
      }
    }

    return true;
  }
};

File: 1897. Redistribute Characters to Make All Strings Equal/solution.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
 var makeEqual = function(words) {
  let dictionary = {};
  for (let index = 0; index < words.length; index++){
    let word = words[index];
    for (let index2 = 0; index2 < word.length; index2++){
      let letter = word[index2];
      dictionary[letter] = (dictionary[letter] || 0) + 1;
    }
  }

  for (let letter in dictionary){
    if (dictionary[letter] % words.length != 0){
      return false;
    }
  }

  return true;
};